src/prefect/deployments/runner.py

```diff
@@ -367,7 +367,9 @@ async def _update(self, deployment_id: UUID, client: PrefectClient):
         await client.update_deployment(
             deployment_id,
             deployment=DeploymentUpdate(
-                **update_payload, parameter_openapi_schema=parameter_openapi_schema
+                **update_payload,
+                parameter_openapi_schema=parameter_openapi_schema,
+                parameters=self.parameters,
```

Comment:
delay serializing the frozen thing

Reply:
a value like `foo` serializes to `['foo']` when we `self.model_dump` the `RunnerDeployment` because of how `BaseAnnotation` works, which means we lose the info that its frozen. We could customize serialization of frozen things using some pydantic trickery, but just letting the new validators handle it felt straightforward to me

Reply:
ah gotcha, makes sense